packages/plugins/typed-variable-modal/test/InputModal.js

```diff
@@ -0,0 +1,58 @@
+
+class InputModal extends TypedVariableModal {
+  /**
+   * The element holding the input type.
+   * @override
+   */
+  createVariableTypeContainer_(types) {
+    const selectDiv = document.createElement('select');
+    for (const type of types) {
+      const displayName = type[0];
+      const typeName = type[1];
+      const option = document.createElement('option');
+      option.classList.add('input-modal-type');
+      option.value = typeName;
+      option.innerText = displayName;
+      selectDiv.appendChild(option);
+    }
+    this.firstTypeInput_ = selectDiv.querySelector('.input-modal-type');
+    return selectDiv;
+  }
+
+  /**
+   * Get the selected type.
+   * @return {string} The selected type.
+   */
+  getSelectedType() {
+    const selectDiv = this.variableTypesDiv_;
```

Comment:
What's the use case for this?

Reply:
In the InputModal I use an Input instead of a list of types to get the selected type. I needed a  way to give the selectedType. Probably doesn't need to be public though, so I can change that.

Reply:
Yeah, I just don't see a use case for this outside of the 'protected' case.